packages/core-transaction-pool-mem/lib/connection.js

```diff
@@ -0,0 +1,340 @@
+'use strict'
+
+const Mem = require('./mem')
+const Storage = require('./storage')
+const container = require('@arkecosystem/core-container')
+const database = container.resolvePlugin('database')
+const emitter = container.resolvePlugin('event-emitter')
+const logger = container.resolvePlugin('logger')
+const { TRANSACTION_TYPES } = require('@arkecosystem/crypto').constants
+const { Transaction } = require('@arkecosystem/crypto').models
+const { TransactionPoolInterface } = require('@arkecosystem/core-transaction-pool')
+
+/**
+ * Transaction pool. It uses a hybrid storage - caching the data
+ * in memory and occasionally saving it to a persistent, on-disk storage (SQLite),
+ * every N modifications, and also during shutdown. The operations that only read
+ * data (everything other than add or remove transaction) are served from the
+ * in-memory storage.
+ */
+class TransactionPool extends TransactionPoolInterface {
+  /**
+   * Make the transaction pool instance. Load all transactions in the pool from
+   * the on-disk database, saved there from a previous run.
+   * @return {TransactionPool}
+   */
+  make () {
+    this.mem = new Mem()
+
+    this.storage = new Storage()
+
+    const allSerialized = this.storage.loadAllInInsertionOrder()
+    allSerialized.map(s => this.mem.add(new Transaction(s), this.options.maxTransactionAge, true))
```

Comment:
Why `.map` instead of `.forEach`?

Reply:
Oh, well, good catch! Changed all in 56a1f0ec. Thanks!